# Exploratory Data Analysis (EDA Basics)Assignment

In [5]:
import pandas as pd

df = pd.read_csv("/content/House Price India.csv")

df.info()

print(df.describe())
df.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 23 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     14620 non-null  int64  
 1   Date                                   14620 non-null  int64  
 2   number of bedrooms                     14620 non-null  int64  
 3   number of bathrooms                    14620 non-null  float64
 4   living area                            14620 non-null  int64  
 5   lot area                               14620 non-null  int64  
 6   number of floors                       14620 non-null  float64
 7   waterfront present                     14620 non-null  int64  
 8   number of views                        14620 non-null  int64  
 9   condition of the house                 14620 non-null  int64  
 10  grade of the house                     14620 non-null  int64  
 11  Ar

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
0,6762810145,42491,5,2.50,3650,9050,2.0,0,4,5,...,1921,0,122003,52.8645,-114.557,2880,5400,2,58,2380000
1,6762810635,42491,4,2.50,2920,4000,1.5,0,0,5,...,1909,0,122004,52.8878,-114.470,2470,4000,2,51,1400000
2,6762810998,42491,5,2.75,2910,9480,1.5,0,0,3,...,1939,0,122004,52.8852,-114.468,2940,6600,1,53,1200000
3,6762812605,42491,4,2.50,3310,42998,2.0,0,0,3,...,2001,0,122005,52.9532,-114.321,3350,42847,3,76,838000
4,6762812919,42491,3,2.00,2710,4500,1.5,0,0,4,...,1929,0,122006,52.9047,-114.485,2060,4500,1,51,805000


In [6]:
# Q1. What is the average, median, and standard deviation of house prices?

# Ans :

df = pd.read_csv("/content/House Price India.csv")

average_price = df['Price'].mean()
median_price = df['Price'].median()
std_price = df['Price'].std()

print("Average House Price:", average_price)
print("Median House Price:", median_price)
print("Standard Deviation of House Price:", std_price)

Average House Price: 538932.2183310534
Median House Price: 450000.0
Standard Deviation of House Price: 367532.3808039669


In [9]:
# Q2. Which number of bedrooms is most common?

# Ans :

df = pd.read_csv("/content/House Price India.csv")

most_common_bedroom = df['number of bedrooms'].mode()[0]

print("Most Common Number of Bedrooms:", most_common_bedroom)

Most Common Number of Bedrooms: 3


In [10]:
# Q3. Check if the price data is skewed.

# Ans :

df = pd.read_csv("/content/House Price India.csv")

price_skewness = df['Price'].skew()

print("Skewness of Price Data:", price_skewness)

Skewness of Price Data: 4.269297720707116


- The price data is highly positively skewed because the skewness value is greater than 1.

**This means :**
- Most houses are priced in the lower-to-middle price range.
- A small number of very expensive luxury houses create a long right tail in the distribution.
- The dataset is not normally distributed.

In [16]:
# Q4. What is the average price for each number of bedrooms?

# Ans :

df = pd.read_csv("/content/House Price India.csv")

avg_bedroom_price = df.groupby('number of bedrooms')['Price'].mean()

print("Average Price for Each Number of Bedrooms:", avg_bedroom_price.round(2))


Average Price for Each Number of Bedrooms: number of bedrooms
1      308963.75
2      398547.57
3      463277.56
4      636198.81
5      775255.03
6      837581.48
7     1016543.90
8     1208454.55
9      776666.33
10     820000.00
11     520000.00
33     640000.00
Name: Price, dtype: float64


- Average price generally increases with the number of bedrooms.
- 7 and 8-bedroom houses have the highest average prices, indicating luxury properties.
- Prices for 9, 10, and 11-bedroom houses do not consistently increase, likely because there are very few such houses in the dataset.
- The 33-bedroom house is highly unusual and likely:
    - a data entry error,
    - incorrect dataset value , or an extreme outlier.

In [19]:
#Q5.  What is the relationship between living area and price? (Use : Area without Basement).

# Ans :

df = pd.read_csv("/content/House Price India.csv")

relationship = df['Area of the house(excluding basement)'].corr(df['Price'])

print("Correlation between Living Area and Price:", relationship)

Correlation between Living Area and Price: 0.6152204198678108


- The correlation value 0.6152 indicates a moderately strong positive relationship between living area and house price.

- This means that as the living area increases, house prices generally tend to increase as well.

- Larger houses usually have higher prices because they provide more usable space and better facilities.
- Since the value is not extremely close to 1, other factors such as location, waterfront, condition, and number of bedrooms also influence house prices.

In [21]:
# Q6. Identify any anomalies where houses have high prices but low areas.

# Ans :

df = pd.read_csv("/content/House Price India.csv")

anomalies = df[
    (df['Price'] > df['Price'].quantile(0.85)) &
    (df['Area of the house(excluding basement)'] < df['Area of the house(excluding basement)'].quantile(0.20))
]

print("Anomalies with High Prices and Low Areas : ","\n ", anomalies)


Anomalies with High Prices and Low Areas :  
                 id   Date  number of bedrooms  number of bathrooms  \
840    6762813159  42505                   3                 2.00   
1210   6762811058  42511                   3                 1.75   
1517   6762811059  42517                   3                 2.50   
1864   6762812455  42523                   3                 1.75   
4458   6762812867  42559                   4                 1.75   
5380   6762812321  42572                   2                 1.75   
6259   6762813150  42585                   4                 1.75   
6796   6762813167  42593                   2                 0.75   
6893   6762812983  42594                   2                 1.00   
7325   6762813204  42601                   3                 1.50   
7747   6762812985  42607                   4                 2.00   
8272   6762812904  42616                   3                 2.00   
8517   6762811560  42621                   3            

- These houses are considered anomalies or outliers.

**Possible reasons:**
- Premium location
- Waterfront/property view
- Luxury interiors
- Historical or commercial value
- Data entry errors

In [23]:
# Q7. Compare average price based on number of floors and houses with or without waterfront.

# Ans :

df = pd.read_csv("/content/House Price India.csv")

comparison = df.groupby(['number of floors', 'waterfront present'])['Price'].mean()

print(comparison.round(2))

number of floors  waterfront present
1.0               0                      432372.39
                  1                     1211080.95
1.5               0                      547616.22
                  1                     1232506.57
2.0               0                      635774.21
                  1                     2075931.37
2.5               0                     1087653.76
                  1                     3280000.00
3.0               0                      555809.50
                  1                     1655000.00
3.5               0                     1205875.00
Name: Price, dtype: float64


- Houses with waterfront views usually have higher average prices.
- Multi-floor houses may also cost more because they generally provide larger living space.
- Waterfront properties are considered premium real estate and attract higher market value.
- Comparing these groups helps identify which property features influence pricing the most.

In [25]:
# Q8. Identify the minimum and maximum house price. What does this indicate?

# Ans :

df = pd.read_csv("/content/House Price India.csv")

min_price = df['Price'].min()
max_price = df['Price'].max()

print("Minimum House Price:", min_price)
print("Maximum House Price:", max_price)

Minimum House Price: 78000
Maximum House Price: 7700000


- The minimum price represents the most affordable property in the dataset (78000).

- The maximum price represents the most expensive or luxury property(7700000).

- A large gap between minimum and maximum prices indicates high variation in the housing market.

**This variation may occur due to:**
- Location differences
- House size
- Waterfront availability
- Luxury features
- Property condition

In [29]:
# Q9. Which location (zipcode/area) has the highest average price?

#  Ans :

df = pd.read_csv("/content/House Price India.csv")

avg_price_location = df.groupby('Postal Code')['Price'].mean()

highest_price_location = avg_price_location.idxmax()
highest_avg_price = avg_price_location.max()

print("Location with Highest Average Price:", highest_price_location)
print("Highest Average Price:", round(highest_avg_price, 2))

Location with Highest Average Price: 122071
Highest Average Price: 2348310.81


**Q10. Write at least 5 insights from your analysis.**

**Ans :** Insights from House Price Analysis

**1. House prices are highly positively skewed**
  - The skewness value was greater than 1 (4.26), indicating that a few luxury houses have extremely high prices compared to most properties.

**2. Living area has a positive relationship with house price**
- The correlation between living area and price was approximately 0.615, showing that larger houses generally cost more.

**3. Houses with more bedrooms usually have higher prices**
- Average house price increased as the number of bedrooms increased, especially for 5+ bedroom houses.

**4. Waterfront houses are more expensive**
- Houses with waterfront views had higher average prices than houses without waterfront access, indicating premium property value.

**5. There are anomalies/outliers in the dataset**
- Some houses had very high prices despite having small living areas.
- The dataset also contained unusual entries like a 33-bedroom house, which may be a data entry error or extreme outlier.

**6. Location strongly affects house prices**
- Certain zipcodes/areas had significantly higher average prices, showing that location is a major factor in real estate valuation.

**7. The housing market contains multiple price segments**
- The large gap between minimum and maximum prices indicates the dataset includes both budget and luxury properties.